### PR TITLE
fix(event): Message event contains stacktrace if `attachStacktrace` o…

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Theoretically only passing the option to the browser client should solve the problem. But for some unexplainable reason, the frames are always an empty array, if the synthetic error is not passed from the RN JS code.

```js
function eventFromString(
  stackParser,
  input,
  syntheticException,
  attachStacktrace,
) {
  const event = {
    message: input,
  };
  let frames;
  if (attachStacktrace && syntheticException) {
    frames = parseStackFrames(stackParser, syntheticException); // 29 frames parsed
    if (frames.length) {
      console.log('frames:', frames, frames.length); //frames: Array(29) 29 Array(29)
      event.exception = {
        values: [{ value: input, stacktrace: { frames } }],
      };
    }
  }

  console.log('event with stack', event, frames);//event with stack Object Array(29)
  //but event.exception.values[0].stacktrace.frames is Array(0)
}
```
Code from JS SDK.
https://github.com/getsentry/sentry-javascript/blob/e1d43ebb34086605aefb9243fce03fc97425b4be/packages/browser/src/eventbuilder.ts#L276

## :bulb: Motivation and Context
Fixes: https://github.com/getsentry/sentry-react-native/issues/2131

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: src/js/sdk.tsx
Comment (krystofwoldrich) [inline src/js/sdk.tsx:288]: When I make this change the `frames` start to disappear and I don't know why.

```suggestion
  return getCurrentHub().captureMessage(
```

Comment (AbhiPrasad) [reply src/js/sdk.tsx:288]: is it because the synthetic exception gets re-written?

Comment (krystofwoldrich) [reply src/js/sdk.tsx:288]: The `synthetic exception` gets all the way to the [`eventFromString`](https://github.com/getsentry/sentry-javascript/blob/e1d43ebb34086605aefb9243fce03fc97425b4be/packages/browser/src/eventbuilder.ts#L276), but there the `frames` are missing in the `stacktrace` even tho that are assigned to it.

**Comment by marandaneto** [comment]
@krystofwoldrich I lack context here, didn't agree on fixing this on JS? https://github.com/getsentry/sentry-javascript/pull/6038

**Comment by krystofwoldrich** [comment]
@marandaneto 

Yes, we will fix the JS part, but besides that, there was an option missing in [RN](https://github.com/getsentry/sentry-react-native/pull/2577/files#diff-7fa63c87605654a66f1a1b0a5463c2b05bc0591ebda5f2e6ed77ea3fbb809674R75).

But after adding [this option](https://github.com/getsentry/sentry-react-native/pull/2577/files#diff-7fa63c87605654a66f1a1b0a5463c2b05bc0591ebda5f2e6ed77ea3fbb809674R75) the stack trace is still missing when `captureMessage` is called thru the hub. The example code from the description shows what goes wrong. But I don't know why.

**Comment by marandaneto** [comment]
> @marandaneto
> 
> Yes, we will fix the JS part, but besides that, there was an option missing in [RN](https://github.com/getsentry/sentry-react-native/pull/2577/files#diff-7fa63c87605654a66f1a1b0a5463c2b05bc0591ebda5f2e6ed77ea3fbb809674R75).
> 
> But after adding [this option](https://github.com/getsentry/sentry-react-native/pull/2577/files#diff-7fa63c87605654a66f1a1b0a5463c2b05bc0591ebda5f2e6ed77ea3fbb809674R75) the stack trace is still missing when `captureMessage` is called thru the hub. The example code from the description shows what goes wrong. But I don't know why.

Gotcha, I'd expect this change to be part of https://github.com/getsentry/sentry-react-native/issues/2461 and not https://github.com/getsentry/sentry-react-native/issues/2131
Why do we have its own `captureMessage`? isn't already defined in the JS SDK?

File: src/js/sdk.tsx
Comment (marandaneto) [reply src/js/sdk.tsx:288]: IDK if its the case, but in some languages/platforms, when you do `Error(message)`, the stack trace isn't available.
You'd need to `throw Error(message)` and only then access the stack trace from the thrown exception.

Comment (krystofwoldrich) [reply src/js/sdk.tsx:288]: The stack is present in any case it just disappears later in the [`eventFromString`](https://github.com/getsentry/sentry-javascript/blob/e1d43ebb34086605aefb9243fce03fc97425b4be/packages/browser/src/eventbuilder.ts#L276)

**Comment by AbhiPrasad** [comment]
This is the weirdest bug I've seen in a loooong time.

As discussed with @krystofwoldrich we probably want to address the bug like so:
1) merge this PR (or if we split this up, it's parts) to unblock users and get it working
2) validate that this is not happening for regular browser JS
  - if it is happening, we need to investigate further on the JS team
3) create a minimal reproduction for react native that just shows the broken functionality and file a ticket with React Native directly to see what is happening

**Comment by marandaneto** [comment]
> This is the weirdest bug I've seen in a loooong time.
> 
> As discussed with @krystofwoldrich we probably want to address the bug like so:
> 
> 1. merge this PR (or if we split this up, it's parts) to unblock users and get it working
> 2. validate that this is not happening for regular browser JS
> 
> * if it is happening, we need to investigate further on the JS team
> 
> 3. create a minimal reproduction for react native that just shows the broken functionality and file a ticket with React Native directly to see what is happening

Ok, that makes sense, I just lacked the context around this and the ingestion.

File: src/js/sdk.tsx
Comment (marandaneto) [reply src/js/sdk.tsx:288]: @krystofwoldrich can you check if this has something to do with `DebugSymbolicator`?
Does it happen in release and debug mode? or only one of them?

Comment (krystofwoldrich) [reply src/js/sdk.tsx:288]: @marandaneto Wow, thank you, DebugSymbolicator is causing it. I haven't looked at why, yet.

**Comment by krystofwoldrich** [comment]
Thank you @AbhiPrasad and @marandaneto 

So the bug is caused by [`DebugSymbolicator`](https://github.com/getsentry/sentry-react-native/blob/f916f2b3afbb9884f9037b8486539ca6cc8b327c/src/js/integrations/debugsymbolicator.ts#L34-L35)

So I think this can be reviewed and merged since in production the stack frames will be sent correctly.

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
> Thank you @AbhiPrasad and @marandaneto
> 
> So the bug is caused by [`DebugSymbolicator`](https://github.com/getsentry/sentry-react-native/blob/f916f2b3afbb9884f9037b8486539ca6cc8b327c/src/js/integrations/debugsymbolicator.ts#L34-L35)
> 
> So I think this can be reviewed and merged since in production the stack frames will be sent correctly.
> 
> For #2461 I will create a new PR that will merge to `5.0.0` branch.

This still depends on the JS SDK bump, after the other PR, right?

**Comment by krystofwoldrich** [comment]
> > Thank you @AbhiPrasad and @marandaneto
> > So the bug is caused by [`DebugSymbolicator`](https://github.com/getsentry/sentry-react-native/blob/f916f2b3afbb9884f9037b8486539ca6cc8b327c/src/js/integrations/debugsymbolicator.ts#L34-L35)
> > So I think this can be reviewed and merged since in production the stack frames will be sent correctly.
> > For #2461 I will create a new PR that will merge to `5.0.0` branch.
> 
> This still depends on the JS SDK bump, after the other PR, right?

The JS SDK bump will move the stack traces from exceptions to threads, but from `sentry.io` user perspective it will start working right away after merging this.

**Pull request merged**
